### PR TITLE
add loader_v4 to builtin list

### PR DESCRIPTION
**Pull request by kbhargava-jump**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -514,6 +514,14 @@ fn load_builtins(cache: &mut ProgramCacheForTxBatch) -> HashSet<Pubkey> {
...
         )),
     );
+    cache.replenish(
+        solana_sdk::loader_v4::id(),
+        Arc::new(ProgramCacheEntry::new_builtin(
+            0u64,
+            0usize,
+            solana_bpf_loader_program::Entrypoint::vm,
+        )),
+    );
```
Comment (mjain-jump) [inline src/lib.rs]: `solana_loader_v4_program::Entrypoint::vm,`

File: src/lib.rs [hunk 2]
```diff
@@ -569,6 +577,7 @@ fn load_builtins(cache: &mut ProgramCacheForTxBatch) -> HashSet<Pubkey> {
     builtins.insert(solana_sdk::bpf_loader::id());
     builtins.insert(solana_sdk::bpf_loader_upgradeable::id());
     builtins.insert(solana_sdk::compute_budget::id());
+    builtins.insert(solana_sdk::loader_v4::id());
```
Comment (mjain-jump) [inline src/lib.rs]: this should be feature gated

Comment (kbhargava-jump) [reply src/lib.rs:580]: pretty sure it's handled with agave here: https://github.com/anza-xyz/agave/blob/024d047eedba22768099cc6509414202400c6231/runtime/src/bank/builtins/mod.rs#L126

**Review by mjain-jump — APPROVED** [review]
note for others - i will be re-adding support for config + alut bpf migration features in a future pr

**Comment by kbhargava-jump** [comment]
added in another PR

**Pull request closed without merging**